internal/workflow/preprocessing.go

```diff
@@ -36,94 +55,225 @@ func (w *PreprocessingWorkflow) Execute(
 		withLocalActOpts(ctx),
 		activities.ValidateMetadataName,
 		&activities.ValidateMetadataParams{MetadataPath: identifySIP.SIP.MetadataPath},
 	).Get(ctx, &validateMetadata)
-
-	// Combine and return validation errors.
-	e = errors.Join(validateStructureErr, validateFileFormatsErr, validateMetadataErr)
 	if e != nil {
-		return nil, e
+		validateMetadataEvent.Complete(
+			ctx,
+			enums.EventOutcomeFailure,
+			"System error: metadata validation has failed",
+		)
+		result.addEvent(validateMetadataEvent)
+		return systemError(logger, "Validate metadata", &result, e)
+	}
+
+	if validateMetadata.Failures != nil {
+		validateMetadataEvent.Complete(
+			ctx,
+			enums.EventOutcomeInvalid,
+			"Content error: metadata validation has failed: %s",
+			strings.Join(validateMetadata.Failures, "\n"),
+		)
+	} else {
+		validateMetadataEvent.Complete(ctx, enums.EventOutcomeValid, "Metadata validation successful")
+	}
+	result.addEvent(validateMetadataEvent)
+
+	// Stop here if the SIP content isn't valid.
+	if !validateStructureEvent.IsValid() ||
+		!validateFileFormatsEvent.IsValid() ||
+		!validateMetadataEvent.IsValid() {
+		return contentError(&result)
 	}
 
 	// Re-structure SIP.
+	restructureSIPEvent := newEvent(ctx, "Restructure SIP")
 	var transformSIP activities.TransformSIPResult
 	e = temporalsdk_workflow.ExecuteActivity(
 		withLocalActOpts(ctx),
 		activities.TransformSIPName,
 		&activities.TransformSIPParams{SIP: identifySIP.SIP},
 	).Get(ctx, &transformSIP)
 	if e != nil {
-		return nil, e
+		result.addEvent(restructureSIPEvent.Complete(
+			ctx,
+			enums.EventOutcomeFailure,
+			"System error: restructuring has failed",
+		))
+		return systemError(logger, "Restructure SIP", &result, e)
 	}
 
+	result.addEvent(restructureSIPEvent.Complete(
+		ctx,
+		enums.EventOutcomeSuccess,
+		"SIP has been restructured",
+	))
+
 	// Bag the SIP for Enduro processing.
+	createBagEvent := newEvent(ctx, "Bag SIP")
 	var createBag bagit.CreateBagActivityResult
 	e = temporalsdk_workflow.ExecuteActivity(
 		withLocalActOpts(ctx),
 		bagit.CreateBagActivityName,
 		&bagit.CreateBagActivityParams{SourcePath: localPath},
 	).Get(ctx, &createBag)
 	if e != nil {
-		return nil, e
+		result.addEvent(createBagEvent.Complete(
+			ctx,
+			enums.EventOutcomeFailure,
+			"System error: bagging has failed",
+		))
+		return systemError(logger, "Create bag", &result, e)
 	}
 
+	result.addEvent(createBagEvent.Complete(
+		ctx,
+		enums.EventOutcomeSuccess,
+		"SIP has been bagged",
+	))
+
 	// TODO: validate checksums located in the XML metadata file
 	// against the checksums generated on Bag creation.
 
-	return &PreprocessingWorkflowResult{RelativePath: params.RelativePath}, e
+	return &result, nil
 }
 
 func withLocalActOpts(ctx temporalsdk_workflow.Context) temporalsdk_workflow.Context {
 	return temporalsdk_workflow.WithActivityOptions(ctx, temporalsdk_workflow.ActivityOptions{
-		ScheduleToCloseTimeout: 5 * time.Minute,
+		ScheduleToCloseTimeout: time.Minute,
```

Comment:
Why this change? Testing leftover? I'd keep it as 5 minutes

Reply:
I changed it because I can't imagine any of the preprocessing activities taking more than a minute to complete.  I'll change it back to 5 minutes though, if you think a minute is too low.